### ISSUE TITLE
Flipbook.play() should return a Future<Event>

### ISSUE THREAD

**Issue by mnordine** [bp74/StageXL#244]
current `play()` returns void. I think it'd be helpful to return a `Future<Event>` here. It would return `_completeEvent`.

Then, you could do something like:

```dart
await ani.play();

// Do next thing
```

as opposed to:

```dart
ani.addEventListener(Event.COMPLETE, doNextThing);
ani.play();

doNextThing(Event e)
{
  // Do next thing
}
```

**Comment by bp74** [comment]
Thanks for the proposal. Getting a future from the play method looks natural. 
But there is also a third option to write the code:

    ani.play();
    await ani.onComplete.first;

The "onComplete" getter returns the Event.Complete stream. 
Every stream has a "first" getter to convert the stream in a future. 

But still, your proposal is the easiest way to write this code. And it's not a breaking change!

**Comment by mnordine** [comment]
Yep, I was actually already doing this in my code. If we return that future it would make it even more readable.

**Comment by bp74** [comment]
I did some changes to the FlipBook class, please take a look at the new method playWith:
https://github.com/bp74/StageXL/blob/master/lib/src/display_ex/flip_book.dart#L96

Would be great if you could give it a try!

**Comment by mnordine** [comment]
lgtm. Only gripe you may get is that some people may expect juggler to be named parameter there as well, i.e.

```dart
ani.playWith(juggler: stage.juggler);
```

Personally, I think it's fine the way you have it, and a nice convenience function.

**Comment by bp74** [comment]
I'm still thinking about the named parameters. This is not only a thing of the Flipbook class but also all over the StageXL code. Have to think about the best way to fix or solve this.
